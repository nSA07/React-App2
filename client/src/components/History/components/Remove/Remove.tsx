import { FC } from 'react'
import { Dot, Disc2  } from 'lucide-react'

import { IHistoryItem } from '@/types/types';

import { formattedData } from '@/helpers/formattedData';


export const Remove: FC<IHistoryItem> = ({prev, next, taskName, dueData}) =>     
{   
    const {createDate} = formattedData(dueData);

    if(!next) {
        return <li data-testid="type-remove" key={dueData} className="text-slate-500 text-xs flex flex-col gap-2">
            <div className="text-slate-500 text-xs flex gap-1 items-center flex-wrap">
                <Dot size={16} />
                You delete <Disc2 size={10} strokeWidth={4} /> <span className="font-bold">{prev}</span> from <span className="font-medium">{taskName}</span>
            </div>
            <span className="pl-5 italic">{createDate}</span>
        </li>;
    } else null;
}