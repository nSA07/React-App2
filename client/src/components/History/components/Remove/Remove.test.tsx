import { render } from '@testing-library/react'

import { Remove } from './Remove'

describe('<Remove />', () => {
    test('should render el with data-testid="type-remove"', async () => {

        const { queryByTestId, container } = render(<Remove prev="prev" next="" taskName="taskName" dueData="1" />)
        
        expect(container.firstChild).toMatchSnapshot()
        expect(queryByTestId('type-added')).toBeDefined()
    })

    test('should NOT render el with data-testid="type-remove"', async () => {

        const { queryByTestId } = render(<Remove prev="prev" next="next" taskName="taskName" dueData="1" />)
        
        expect(queryByTestId('type-added')).toBeFalsy()
    })
})