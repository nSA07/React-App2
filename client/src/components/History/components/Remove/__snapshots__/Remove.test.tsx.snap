// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Remove /> should render el with data-testid="type-remove" 1`] = `
<li
  class="text-slate-500 text-xs flex flex-col gap-2"
  data-testid="type-remove"
>
  <div
    class="text-slate-500 text-xs flex gap-1 items-center flex-wrap"
  >
    <svg
      class="lucide lucide-dot "
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12.1"
        cy="12.1"
        r="1"
      />
    </svg>
    You delete 
    <svg
      class="lucide lucide-disc2 "
      fill="none"
      height="10"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="4"
      viewBox="0 0 24 24"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M12 12h.01"
      />
    </svg>
     
    <span
      class="font-bold"
    >
      prev
    </span>
     from 
    <span
      class="font-medium"
    >
      taskName
    </span>
  </div>
  <span
    class="pl-5 italic"
  >
    Jan 1, 12:00 AM
  </span>
</li>
`;
